stream_alert/rule_processor/payload.py

```diff
@@ -331,7 +332,19 @@ def pre_parse(self):
             self.raw_record['Sns']['MessageId'],
             self.raw_record['EventSubscriptionArn'])
 
-        self.pre_parsed_record = self.raw_record['Sns']['Message']
+        # check if SNS message is an Amazon S3 Notification type
+        if 'Subject' in self.raw_record['Sns'] and \
+            self.raw_record['Sns']['Subject'] == "Amazon S3 Notification":
+            LOGGER.debug('Amazon S3 notification detected.. parsing message')
+
+            message = json.loads(self.raw_record['Sns']['Message'])['Records'][0]
+            # encapsulate in s3 stream payload
+            s3_payload = load_stream_payload('s3', 'sns-passthrough', message)
```

Comment:
This is a neat trick!! I would still suggest actually passing the instance's `entity` to the `load_stream_payload` for consistency about where the data came from:

`s3_payload = load_stream_payload('s3', self.entity, message)`